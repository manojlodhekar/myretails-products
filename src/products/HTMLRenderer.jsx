import React from 'react';

const HTMLRenderer = (props) =>{
    return(
        <div>
            <div style={{paddingTop: 10}}>
                <span id="htmlRendererId" className="returnPolicyCls" dangerouslySetInnerHTML={{__html: props.htmlContents}} />
            </div>
        </div>
    )
}
export default HTMLRenderer;