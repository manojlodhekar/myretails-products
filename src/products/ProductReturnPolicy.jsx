import React from 'react';
import HTMLRenderer from './HTMLRenderer';

const ProductReturnPolicy = (props) =>{

    const renderProductReturnPolicy = () =>{

        let returnPolicy =  props.returnPolicyDetails.ReturnPolicyDetails.map((policy, index) => {
            return(
                <div className="returnPolicyCls" key={index}> 
                    { 
                        index === 0 ? <div className="policyHeader">{policy.guestMessage} : </div>  : null
                    }
                    
                    <div>
                    {policy.user} : {policy.policyDays} Days
                    </div>
                </div>
            )
        });
        return returnPolicy;
    }

    return(
        <div>
            {renderProductReturnPolicy()}
            <div style={{paddingTop: 10}}>
                <HTMLRenderer htmlContents={props.returnPolicyDetails.legalCopy} />
            </div>
        </div>
    )
}
export default ProductReturnPolicy;