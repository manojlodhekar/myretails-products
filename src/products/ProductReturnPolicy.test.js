import React from 'react';
import ReactDOM from 'react-dom';
import ProductReturnPolicy from './ProductReturnPolicy';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount } from 'enzyme';
import HTMLRenderer from './HTMLRenderer';

jest.mock('react-dom');

Enzyme.configure({ adapter: new Adapter() });
let prodReturn;
const itemPayload = {"ReturnPolicy": [
  {
   "ReturnPolicyDetails": [
    {
     "guestMessage": "View our return policy",
     "policyDays": "100",
     "user": "Regular Guest"
    },
    {
     "guestMessage": "View our return policy",
     "policyDays": "120",
     "user": "Best Guest"
    }
   ]}]}
beforeEach(()=>{
  prodReturn = mount(<ProductReturnPolicy returnPolicyDetails={itemPayload}>
  <HTMLRenderer  htmlContents="abcd"/>
  </ProductReturnPolicy>);

})
describe('ProductReturnPolicy is defined', () => {

  it('should render correctly', () => {
    expect(prodReturn).toHaveLength(1);
  });

  it('should render inner component', () => {
    expect(prodReturn).toMatchSnapshot();
  });

  it('should render curent policy props of component', () => {
    prodReturn = shallow(<ProductReturnPolicy returnPolicyDetails={itemPayload.ReturnPolicy[0]}/>);// Best Guest : 120 Days
    expect(prodReturn.find('.returnPolicyCls').last().text()).toEqual(itemPayload.ReturnPolicy[0].ReturnPolicyDetails[1].user + ' : ' +  itemPayload.ReturnPolicy[0].ReturnPolicyDetails[1].policyDays + ' Days');
  });
  it('should render retunr policy of component', () => {
    prodReturn = shallow(<ProductReturnPolicy returnPolicyDetails={itemPayload.ReturnPolicy[0]}/>);
    //console.log(itemPayload.ReturnPolicy[0].ReturnPolicyDetails[0].guestMessage);
     expect((prodReturn.find('.policyHeader')).text()).toEqual(itemPayload.ReturnPolicy[0].ReturnPolicyDetails[0].guestMessage + ' : ');
  });



 });
