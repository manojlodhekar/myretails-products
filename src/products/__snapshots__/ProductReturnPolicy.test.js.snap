// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductReturnPolicy is defined should render inner component 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <ProductReturnPolicy
    returnPolicyDetails={
        Object {
            "ReturnPolicy": Array [
              Object {
                "ReturnPolicyDetails": Array [
                  Object {
                    "guestMessage": "View our return policy",
                    "policyDays": "100",
                    "user": "Regular Guest",
                  },
                  Object {
                    "guestMessage": "View our return policy",
                    "policyDays": "120",
                    "user": "Best Guest",
                  },
                ],
              },
            ],
          }
    }
>
    <HTMLRenderer
        htmlContents="abcd"
    />
</ProductReturnPolicy>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): null,
  Symbol(enzyme.__nodes__): Array [
    null,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
